transcode/transcode.go

```diff
@@ -107,36 +108,35 @@ func RunTranscodeProcess(transcodeRequest TranscodeSegmentRequest, streamName st
 				CustomAPIURL: transcodeRequest.TranscodeAPIUrl,
 			}
 			broadcasterClient, _ := clients.NewRemoteBroadcasterClient(creds)
-
-			tr, err := broadcasterClient.TranscodeSegmentWithRemoteBroadcaster(rc, int64(segmentIndex), transcodeProfiles, streamName, u.DurationMillis)
+			// Get renditions from remote broadcaster
+			tr, err = broadcasterClient.TranscodeSegmentWithRemoteBroadcaster(rc, int64(segmentIndex), transcodeProfiles, streamName, u.DurationMillis)
 			if err != nil {
 				return fmt.Errorf("failed to run TranscodeSegmentWithRemoteBroadcaster: %s", err)
 			}
-			fmt.Println("transcodeResult", tr) //remove this
-			// TODO: Upload the output segments
 		} else {
-			tr, err := localBroadcasterClient.TranscodeSegment(rc, int64(segmentIndex), transcodeProfiles, u.DurationMillis)
+			// Get renditions from local broadcaster
+			tr, err = localBroadcasterClient.TranscodeSegment(rc, int64(segmentIndex), transcodeProfiles, u.DurationMillis)
 			if err != nil {
 				return fmt.Errorf("failed to run TranscodeSegment: %s", err)
 			}
+		}
+		// Store renditions
+		for _, transcodedSegment := range tr.Renditions {
+			renditionIndex := getProfileIndex(transcodeProfiles, transcodedSegment.Name)
+			if renditionIndex == -1 {
+				return fmt.Errorf("failed to find profile with name %q while parsing rendition segment", transcodedSegment.Name)
+			}
 
-			for _, transcodedSegment := range tr.Renditions {
-				renditionIndex := getProfileIndex(transcodeProfiles, transcodedSegment.Name)
-				if renditionIndex == -1 {
-					return fmt.Errorf("failed to find profile with name %q while parsing rendition segment", transcodedSegment.Name)
-				}
-
-				relativeRenditionPath := fmt.Sprintf("rendition-%d/", renditionIndex)
-				relativeRenditionURL, err := url.Parse(relativeRenditionPath)
-				if err != nil {
-					return fmt.Errorf("error building rendition segment URL %q: %s", relativeRenditionPath, err)
-				}
-				renditionURL := targetOSURL.ResolveReference(relativeRenditionURL)
-
-				err = clients.UploadToOSURL(renditionURL.String(), fmt.Sprintf("%d.ts", segmentIndex), bytes.NewReader(transcodedSegment.MediaData))
-				if err != nil {
-					return fmt.Errorf("failed to upload master playlist: %s", err)
-				}
+			relativeRenditionPath := fmt.Sprintf("rendition-%d/", renditionIndex)
+			relativeRenditionURL, err := url.Parse(relativeRenditionPath)
+			if err != nil {
+				return fmt.Errorf("error building rendition segment URL %q: %s", relativeRenditionPath, err)
+			}
+			renditionURL := targetOSURL.ResolveReference(relativeRenditionURL)
+
+			err = clients.UploadToOSURL(renditionURL.String(), fmt.Sprintf("%d.ts", segmentIndex), bytes.NewReader(transcodedSegment.MediaData))
+			if err != nil {
+				return fmt.Errorf("failed to upload master playlist: %s", err)
```

Comment:
Diff looks funny in github, this is actually same code with decreased indent

Reply:
You can turn off whitespace in the Github diff viewer with https://github.com/livepeer/catalyst-api/pull/112/files?diff=unified&w=1